pages/Identity/Registry.md

```diff
@@ -6,21 +6,228 @@ menu: Identity
+while maintaining all graph connections, public and private.
+
 ## Specification Status
 
 | Version | Status |
 ---------- | ---------
-| 0.1     | Draft |
+| 0.1     | Tentative |
 
 ## Purpose
-1. Describe how an Identity Registry resolves a name or number to an identity contract
+1. Describes how the Identity Registry resolves a DSNP Id to an identity contract address
+1. Describes how the Identity Registry allows for handle resolution
+1. Presents the interface for the Identity Registry
+1. Describes rejected alternatives
 
 ## Assumptions
 
-TBD
+* Ids will need to be moved from Betanet to Mainnet
+* Handles are for display and discovery purposes only
+
+## Discovery via DSNP Handles
+
+Contract addresses or numerical ids are not easy to remember.
+Most networks rely on text based handles for discovery of users on a network.
+DSNP handles are an easy way to allow easy user lookup.
+
+### Handles
+
+Handles are simple UTF-8 strings.
+No limitations are placed on length or contents, although different clients may not have support for the full unicode set.
+A user MAY NOT register multiple handles that point to the same DSNP Id.
+Handles must be unique.
+
+### Homograph Attack Mitigation
+
+UTF-8 support for handles opens handle users up to [homograph attacks](https://en.wikipedia.org/wiki/IDN_homograph_attack), not to mention case-sensitivity issues.
+This issue is of ongoing discussion both for the DSNP as well as in for ICANN domain names and other projects working with internationalization support.
+
+Because the DSNP Id is stable, attacks would only be successful in cases where the DSNP Id were unknown.
+[Punycode](https://en.wikipedia.org/wiki/Punycode) would resolve some issues, but would require losing true internationalization.
+
+#### Current Mitigation Strategies
+
+Clients resolving handles MUST implement a method to detect potential homographs and check both user settings and potentially check the registry for additional potential matching DSNP Ids.
+
+## DSNP Ids
+
+Ethereum contract addresses are currently 160 bit values which is much larger than needed for unique identification.
+Identification can be reduced to just 64 bit identifiers with the registry and enable contract changing.
+
+**Remember:** Only DSNP Ids are safe for long term data connections.
+
+* DSNP Identity contract addresses are not guaranteed long term for upgrade and migration reasons
+* Handles can change
+
+## Resolutions
+
+Resolutions are possible between any of the three pieces of data: Handle, DSNP Id, and Contract Address.
+While a utility method is provided for ease of moving from handle to the contract address,
+the other resolutions require using contract log events.
+
+### Current Handle -> Current Contract Address
+
+The `IRegistry.resolveHandleToAddress` method is the most efficient for discovery of current values.
+Perform a log search using the `DSNPId` event to discover all contract addresses and DSNP Ids that have had a given handle. 
```

Comment:
this line and line 79 say the same thing. Are both of these functions returning DSNP ids and contract addresses. It seems like one should resolve to ID and one should resolve to contract address?

Reply:
That isn't super clean. How does this sound:

> To retrieve historical values, perform a log search using the `DSNPRegistryUpdate` event.

Or we could just remove them. The section "Other Lookups & Historical Values" has the information. The key is that that event has all three pieces of data, so it can be used to perform lookups in any direction.

Reply:
Yes, I think what you suggested sounds better